src/DurableTask.Netherite/OrchestrationService/NetheriteOrchestrationService.cs

```diff
@@ -88,58 +103,82 @@ public override string ToString()
 #else
             string configuration = "Release";
 #endif
-            return $"NetheriteOrchestrationService on {this.configuredTransport}Transport and {this.configuredStorage}Storage, {configuration} build";
+            return $"NetheriteOrchestrationService on {this.Settings.TransportChoice}Transport and {this.Settings.StorageChoice}Storage, {configuration} build";
         }
 
         /// <summary>
         /// Creates a new instance of the OrchestrationService with default settings
         /// </summary>
         public NetheriteOrchestrationService(NetheriteOrchestrationServiceSettings settings, ILoggerFactory loggerFactory)
+            : this(settings, loggerFactory, null)
+        {
+        }
+        
+        /// <summary>
+        /// Creates a new instance of the OrchestrationService with default settings
+        /// </summary>
+        public NetheriteOrchestrationService(NetheriteOrchestrationServiceSettings settings, ILoggerFactory loggerFactory, IServiceProvider serviceProvider)
```

Comment:
As with the `NetheriteProviderFactory`, I recommend using DI/IoC over service-locator.

Reply:
I understand that the service locator may not always be the preferred solution, but I think it is appropriate in this situation.

- Not all of the services are public, so it is not just a simple case of listing them in the constructor argument. 
- Constructors that list all the dependencies create maintenance complexity. In some sense, this is the problem DI is solving in the first place so it seems sad to give up on that. Also, I really dislike having lots of overloads, it makes the code hard to read.
 
Using a service locator instead helps to keep the constructor stable because new dependencies can be found without having to be listed explicitly. 

From what I understand, the danger of the service locator pattern is that the dependency ordering is not visible to the DI startup logic, so it cannot correctly order dependencies during startup. But that is not an issue here because the orchestration service is never constructed at that point. Only the factories are constructed at startup time.